tests/db_functions/datetime/test_extract_trunc.py

```diff
@@ -1098,6 +1098,41 @@ def test_trunc_subquery_with_parameters(self):
             ]
         )
 
+    def test_extract_outerref(self):
+        def dt(*args):
+            dt = datetime(*args)
+            return timezone.make_aware(dt, is_dst=False) if settings.USE_TZ else dt
+
+        DTModel.objects.create(name='extract_outerref_1', start_datetime=dt(2000, 1, 1))
+        DTModel.objects.create(name='extract_outerref_2', start_datetime=dt(2001, 2, 5))
+        DTModel.objects.create(name='extract_outerref_3', start_datetime=dt(2002, 1, 7))
+        DTModel.objects.create(name='extract_outerref_4', start_datetime=dt(2003, 2, 8))
+
+        subquery_qs = DTModel.objects.filter(
+            name__in=['extract_outerref_3', 'extract_outerref_4'],
+            start_datetime__month=ExtractMonth(OuterRef('start_datetime')),
+        )
+
+        qs = DTModel.objects.filter(
+            name__in=['extract_outerref_1', 'extract_outerref_2']
+        ).annotate(
+            related_name=Subquery(subquery_qs.values('name')[:1]),
+        )
+        values_list = list(qs.order_by('name').values_list('name', 'related_name'))
+        self.assertEqual(values_list, [
+            ('extract_outerref_1', 'extract_outerref_3'),
+            ('extract_outerref_2', 'extract_outerref_4'),
+        ])
+
+    def test_extract_outerref_validation(self):
+        subquery_qs = DTModel.objects.filter(name=ExtractMonth(OuterRef('name')))
```

Comment:
Thanks for adding a test that proves that `ResolvedOuterRef.resolve_expression` is eventually called and checked against.